pkg/awsutils/awsutils.go

```diff
@@ -1296,16 +1384,42 @@ func (cache *EC2InstanceMetadataCache) waitForENIAndIPsAttached(eni string, want
 }
 
 // DeallocIPAddresses allocates numIPs of IP address on an ENI
-func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []string) error {
+func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []string, ignoreCachedPDflag bool) error {
```

Comment:
this `ignoreCachedPDflag` is not clear from interface design. (i can understand it means ignore the ipv4PrefixDelegation setting).
We should change this into two separate function for ipv4 address and for prefix

Reply:
The reason I didn't separate the functions was because we need to duplicate the functions expect for the input and if any fixes, we need to make sure it goes in 2 places. Hence I added the check at the last step to keep the changes minimal.